prime-router/src/main/kotlin/Report.kt

```diff
@@ -415,11 +429,32 @@ class Report : Logging {
             filteredTable,
             fromThisReport("filter: $filterFunctions")
         )
-        filteredReport.filteredItems.addAll(filteredRows)
+        filteredReport.filteringResults.addAll(this.filteringResults) // copy ReportStreamFilterResults from prev
+        filteredReport.filteringResults.addAll(filteredRows) // and add any new ReportStreamFilterResults just created.
         filteredReport.itemLineages = createItemLineages(finalCombinedSelection, this, filteredReport)
         return filteredReport
     }
 
+    /**
+     * Return the values in column [columnName] for these [rows].  If a [default] is specified,
+     * that value is used for all cases where the value is empty/null/missing for that row in the table.
+     * If [default] is null, this may return fewer rows than in [rows].
+     *
+     * @return an emptyList if the columnName is null (not an error)
+     *
+     */
+    fun getValuesInRows(columnName: String?, rows: List<Int>, default: String? = null): List<String> {
```

Comment:
This helper function is general, but created for the specific purpose of pulling out the trackingElement IDs for the subset of rows that have been filtered out, so we can delivery good logging and error messages.